source/dub/dependency.d

```diff
@@ -114,47 +112,125 @@ struct Dependency {
 		this(repository);
 	}
 
+	/// Allow to act on the `Dependency` based on its type
+	public auto visit (H1, H2, H3) (scope H1 h1, scope H2 h2, scope H3 h3) inout
```

Comment:
Right. That was mostly an experiment, as my guess was that all places will use it as a `string`, and will interact with `Dependency` as if it was a `VersionRange`. I agree with you that making `VersionRange` public is the way to go.
W.r.t. visit, perhaps it would be best to just go with fixed order. TBH at first it was just an alias but that didn't work so well, and then it kinda grew on its own :)
Will revisit soon.